### PR TITLE
Alpha/nested delete/bug

### PR DESCRIPTION
**Pull request by AlphaOD**
- Delete Row didn't work for nested grids. 

Now a new function called Reassign has been added to recursively reassign every level of nested grids in case of delete. _**(still needs work for column and styling)**_

The problem : 
Delete

Solved:
deleteRow

- AddnestedGrid didn't work for anything different from the Root grid, because the next line replaced the root grid or the selected cell's parents' grid with the selected cell's grid.
carbon

### CODE REVIEW

File: src/model.rs [hunk 1]
```diff
@@ -283,6 +283,44 @@ impl Model {
             })
             .collect()
     }
+    fn Reassign(&mut self, Coord: Coordinate, mut grammars: HashMap<Coordinate, Grammar>, i: i32) -> HashMap<Coordinate, Grammar> {
```
Comment (Korede-TA) [inline src/model.rs:286]: could you make this name lowercase? also, i don't think you need to pass in both the model and the grammars
```suggestion
    fn reassign(&mut self, Coord: Coordinate, mut grammars: HashMap<Coordinate, Grammar>, i: i32) -> HashMap<Coordinate, Grammar> {
    let grammars = self.get_session().grammars;
```

File: src/model.rs [hunk 2]
```diff
@@ -700,34 +728,38 @@ impl Component for Model {
             }
 
             Action::ReadSession(file) => {
-                let callback = self.link.callback(Action::LoadSession);
-                let task = self.reader.read_file(file, callback);
-                self.tasks.push(task);
+                info!("in here");
```
Comment (Korede-TA) [inline src/model.rs:731]: remember to take out these log statements before merging!!

File: src/model.rs [hunk 3]
```diff
@@ -1100,22 +1127,30 @@ impl Component for Model {
                     //Changing each rowfrom the one being deleted
                     while let Some(below_coord) = next_row.neighbor_below() {
                         temp.clear();
-                        row_coords2 = self.query_row(below_coord.full_row());
-                        if row_coords2.len() != 0 {
+                        next_row_coord = self.query_row(below_coord.full_row());
```
Comment (Korede-TA) [inline src/model.rs:1130]: good idea to rename these variables! makes it more obvious what they are

File: src/model.rs (hunk 2 above)
Comment (AlphaOD) [reply src/model.rs:836]: I worked on my save-loading branch that's why there is this comment still, I ll merge this before that one so it should be fine

**Pull request closed without merging**